examples/puffin_profiler/Cargo.toml

```diff
@@ -28,3 +30,4 @@ env_logger = { version = "0.10", default-features = false, features = [
 log = { workspace = true }
 puffin = "0.19"
 puffin_http = "0.16"
+profiling = {workspace = true, features = ["profile-with-puffin"] }
```

Comment:
Always having the profile-with-puffin feature enabled means egui will now always require puffin, which isn't great. I think instead we should keep the puffin features in the crates and make them enable the `profiling/profile-with-puffin` feature. 

Alternatively we could remove the puffin features and users could just enable the profiling implementation they'd like by enabling the relevant profiling feature themselves.

Also, I don't think we're using the proc macro, so we should set default-features to false.

Reply:
Note that this cargo.toml is just in the puffin_profiler sample, based on the name it seems pretty reasonable to require it I think. For egui in general default features are disabled and no profiler is enabled by default I believe

Reply:
I just double checked regarding 'default-features', it's set to false in the workspace, so I don't think it should be necessary to put that in individual crates when they inherit settings from the workspace.

Reply:
Ooops, I totally missed that this was a example